salsa_auth/views.py

```diff
@@ -47,22 +47,39 @@ class SignUpForm(JSONFormResponseMixin, FormView):
     template_name = 'signup.html'
 
     def form_valid(self, form):
-        user = self._make_user(form.cleaned_data)
+        email = form.cleaned_data['email']
 
-        # TO-DO: Potentially intercept SMTP error for undeliverable mail here
-        self._send_verification_email(user)
+        # If the email already exists in Salsa, re-verification is not required.
+        # Authenticate the user.
+        if salsa_client.get_supporter(email):
+            return redirect('salsa_auth:authenticate')
 
-        messages.add_message(self.request,
-                             messages.INFO,
-                             'Thanks for signing up!',
-                             extra_tags='font-weight-bold')
+        # If the given email is in Salsa, and that email has hard-bounced,
+        # inform the user, and do not send another email to the invalid address.
+        if salsa_client.get_supporter(email, allow_invalid=True):
+            message_title = "Sorry we missed you!"
+            message_body = 'We could not deliver mail to <strong>{0}</strong>. Please supply a working email address.'.format(user.email)
 
-        messages.add_message(self.request,
-                             messages.INFO,
-                             'Please check your email for an activation link.')
+        user = User.objects.filter(email=email).order_by('date_joined').first()
+
+        if not user:
+            user = self._make_user(form.cleaned_data)
+
+            self._send_verification_email(user)
+
+            message_title = 'Thanks for signing up!'
+            message_body = 'Please check your email for an activation link.'
+
+        else:
+            message_title = 'You look familiar!'
```

Comment:
Related to the above comment, change this to "Please verify this email" or something similar